### ISSUE TITLE
Migrate SBA to Gradle build tool

### ISSUE THREAD

**Issue by arunkpatra** [codecentric/spring-boot-admin#1438]
The venerable Ant build tool, hugely customisable with its custom tasks, finally gave way to the modern Maven ecosystem. That's natural evolution. New needs, new tools. History repeats, and Gradle builds upon all the goodness of Maven and includes many advanced (and more importantly useful) features like build-cache, parallel execution, Gradle daemons etc. Many major OSS vendors have switched after tasting the goods, Spring Boot included.

This enhancement request is to migrate SBA to a Gradle based build system.

Would be glad to submit a PR when you want to migrate.

**Comment by joshiste** [comment]
Personally I'm not that much used to gradle and prefer maven. Also the comparison of maven is to ant as gradle is to maven imho doesn't work.

Migrating to grade is a lot of effort as automated build and release process (including signing of the artifacts) would be affected.

Could you elaborate a bit on the advantages we would have using gradle as I'm currently not convinced if we should take this step.

**Comment by arunkpatra** [comment]
IMHO, if we look from the perspective of "What's the cost of not doing it?" the answer is clearly a zero in the perceivable future. Again, from the perspective of "If it's not broken, why fix it?" the answer is clearly to stay with Maven. I'm with you on that, and moreso from the perspective of being the project owner and your own familiarity with Maven. Finally, we are talking about the build tool here and not SBA per say, so no real need to jump ship.

## tl;dr

- [x] SBA has gained good traction in the market, people love it, and now SBA going cloud native into the K8s world, its natural that interest in SBA would go north. More people are expected to take interest into the roadmap of SBA and contribute.
- [x] Future contributors to SBA in are expected to have a natural sway towards Gradle. But honestly the build tool itself does not matter. But then I don't use Ant or CVS for my new projects - not a direct comparison though.
- [x] Now from a SBA developer perspective I like my builds to be blazing fast, let's say sometimes 10% of my current Maven builds - I would love that. Build cache, parallel builds all hell towards this. Scan reports are awesome. The HTML dependency graph is awesome. Generating jacoco aggregate reports is a breeze. Gradle plays fully nicely with Maven - no heartache with Maven at all. Every single artifact produced is 100% compatible with Maven built artifacts.
- [x] I would love to see superior transitive dependency management and not allow dependencies to leak into the caller. Not that SBA leaks dependencies right now. But Gradle's approach is much superior when it comes to transitive dependency management.
- [x] The Spring Boot ecosystem itself has embraced Gradle wholeheartedly - not that SBA needs to toe their line, but I would think @dsyer did give it a good hard look before bringing in Gradle - I haven't talked to Dave though on this yet.
- [x] A very rich set of equivalent and more modern build plugins of Maven plugins are already available.

### Challenges with the migration
1. Codebase structure might have to change slightly
2. Build and automation need to change.

### The rewards of migration
1. Be future ready
2. Build speed enhancements for SBA
3. Adopt a superior build tool

Any takers?

_(PS: Disclaimer - I'm not associated with the Gradle project or Gradle Inc. in any way whatsoever)_
